src/CommunityToolkit.Maui.MediaElement/Services/MediaControlsService.android.cs

```diff
@@ -0,0 +1,317 @@
+﻿using System.ComponentModel;
+using System.Runtime.InteropServices;
+using Android.App;
+using Android.Content;
+using Android.Content.PM;
+using Android.Media;
+using Android.OS;
+using Android.Support.V4.Media.Session;
+using AndroidX.Core.App;
+using AndroidX.LocalBroadcastManager.Content;
+using CommunityToolkit.Maui.Core.Views;
+using Microsoft.Win32.SafeHandles;
+using Resource = Microsoft.Maui.Resource;
+using Stream = Android.Media.Stream;
+
+namespace CommunityToolkit.Maui.Media.Services;
+
+[Service(Exported = true,Enabled = true, Name = "CommunityToolkit.Maui.Services", ForegroundServiceType = ForegroundService.TypeMediaPlayback)]
+public class MediaControlsService : Service	
+{
+	bool disposedValue;
+	public const string ACTION_PLAY = "MediaAction.play";
+	public const string ACTION_PAUSE = "MediaAction.pause";
+	public const string ACTION_FORWARD = "MediaAction.advance_forward_30";
+	public const string ACTION_BACK = "MediaAction.advance_backward_10";
+	public const string ACTION_UPDATE_UI = "CommunityToolkit.Maui.Services.action.UPDATE_UI";
+	bool notificationExists = false;
+	PendingIntentFlags pendingIntentFlags;
+
+	SafeHandle? safeHandle = new SafeFileHandle(IntPtr.Zero, true);
+	MediaSessionCompat? mediaSession;
+	AudioManager? audioManager;
+	NotificationCompat.Builder? notification;
+	NotificationCompat.Action? actionPlay;
+	NotificationCompat.Action? actionPause;
+	NotificationCompat.Action? actionNext;
+	NotificationCompat.Action? actionPrevious;
+	MediaSessionCompat.Token? token;
+	ReceiveUpdates? receiveUpdates;
+
+	public MediaControlsService()
+	{
+	}
+
+	public override IBinder? OnBind(Intent? intent)
+	{
+		return null;
+	}
+
+	public override StartCommandResult OnStartCommand(Intent? intent, StartCommandFlags flags, int startId)
+	{
+		ArgumentNullException.ThrowIfNull(intent);
+		switch (intent.Action)
+		{
+			case "MediaAction.pause":
+				HandleNotificationEvent(false);
+				BroadcastUpdate(ACTION_PAUSE);
+				break;
+			case "MediaAction.play":
+				HandleNotificationEvent(true);
+				BroadcastUpdate(ACTION_PLAY);
+				break;
+			case "MediaAction.previous":
+				HandleNotificationEvent(true);
+				BroadcastUpdate(ACTION_BACK);
+				break;
+			case "MediaAction.next":
+				HandleNotificationEvent(true);
+				BroadcastUpdate(ACTION_FORWARD);
+				break;
+		}
+
+		_ = startForegroundServiceAsync(intent);
```

Comment:
Do we want to give feedback if it fails?

Reply:
It will fail on Android at specific API levels. I changed it to use `.ConfigureAwait(false)` so that it fails gracefully and does not deadlock. Not all Android API's support this feature. Those that do require it and it is not optional.